future.go

```diff
@@ -46,6 +48,15 @@ type ConfigurationFuture interface {
 	Configuration() Configuration
 }
 
+// SnapshotFuture is used for waiting on a user-triggered snapshot to complete.
+type SnapshotFuture interface {
+	Future
+
+	// Open is a function you can call to access the underlying snapshot and
+	// its metadata.
+	Open() (*SnapshotMeta, io.ReadCloser, error)
```

Comment:
Is it only safe to call this once we get a non-nil Err back? Good to just add to the comment.